homeassistant/components/hive/binary_sensor.py

```diff
@@ -56,9 +76,10 @@ def device_state_attributes(self):
     @property
     def is_on(self):
         """Return true if the binary sensor is on."""
-        return self.session.sensor.get_state(self.node_id, self.node_device_type)
+        return self.device["status"]["state"]
 
-    def update(self):
+    async def async_update(self):
         """Update all Node data from Hive."""
-        self.session.core.update_data(self.node_id)
-        self.attributes = self.session.attributes.state_attributes(self.node_id)
+        await self.hive.session.updateData(self.device)
+        self.device = await self.hive.sensor.get_sensor(self.device)
+        self.attributes = self.device.get("attributes", {})
```

Comment:
Please set the state attribute keys explicitly in the attributes. We want to see what they are and their formatting.

Eg create a constant dict that maps between state attribute key and device attribute key. Then iterate over that dict and lookup the device values.